crypto/crypto/src/noise.rs

```diff
@@ -121,7 +121,7 @@ pub const fn handshake_resp_msg_len(payload_len: usize) -> usize {
 
 /// This implementation relies on the fact that the hash function used has a 256-bit output
 #[rustfmt::skip]
-const _: [(); 0 - !{ const ASSERT: bool = HashValue::LENGTH == 32; ASSERT } as usize] = [];
+const _: [(); 32] = [(); HashValue::LENGTH];
```

Comment:
I'm not sure how this ended up in this PR, but as far as I see this is just an assertion that the hash value length is always 32?  I will say that the result is definitely easier to understand :D

Reply:
ahh, this slipped in, though i'm probably going to leave it : )

i was looking at common sizes of various network structs (that we send around in libra-channels). for example

```
size_of<(PeerId, ProtocolId)>() == 17 bytes
size_of<PeerManagerRequest>() == 88 bytes
size_of<PeerManagerNotification>() == 72 bytes
size_of<(PeerManagerRequest, Option<oneshot::Sender<ElementStatus<PeerManagerRequest>>>)>() == 96 bytes
```

Reply:
If you can, put it in another commit.